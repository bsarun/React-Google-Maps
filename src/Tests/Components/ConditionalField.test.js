import React from 'react';
import ConditionalField from '../../Components/ConditionalField';
import { IntlProvider } from 'react-intl';
import i18n from '../../i18n.en.json';
import renderer from 'react-test-renderer';
import App from '../../Containers/App.jsx';
import {mount, shallow} from 'enzyme';
import sinon from 'sinon';
import PropTypes from 'prop-types';
import PortalAuthActions from '../../Actions/PortalAuthActions';
jest.mock('../../Actions/PortalAuthActions');
import AppDispatcher from '../../Dispatchers/AppDispatcher';
jest.mock('../../Actions/EndpointActionUtils');

let props, output, tree;

describe('OptIn', () => {
	beforeEach(() => {
	props = {
		location: { pathname: 'Opt-in', query: {pendingAppId: 999999}},
		intl: {
			messageIfExists : jest.fn()
		},
		condition:11,
		personId:123,
		hiddenFields:[{
			fieldname:11
		}],
		dynamicQuestions :{
			persons:[{
				personId:123,
				hiddenFields:[0]
			}]
		},
		dynamicQuestionId:0
	} 
	Object.defineProperty(window, "matchMedia", {
		value: jest.fn(() => { return { matches: true } })
	});
	
let person = {
	personId:123
};
const dynamicQuestionsForPerson = {
	persons:{
			find:jest.fn(() => { return { matches: true } })
	},
	outputglobalresname:['CP1'],
	outputglobalresvalue:['1']
};
	
	output = renderer.create(<IntlProvider locale="en" messages={i18n} >
                                      <App {...props}>
                                        <Formsy.Form>
                                          <ConditionalField dynamicQuestionsForPerson={dynamicQuestionsForPerson}
																					person={person}
																					dynamicQuestions={props.dynamicQuestions}/>
                                        </Formsy.Form>
                                      </App>
                                    </IntlProvider>);
	tree = output.toJSON();
});	

	it('Testing conditional field', () => {
		expect(tree).toMatchSnapshot();
	});
	it('Testing findOnePerson function', () => {
		const dynamicQuestionsForPerson = {
			persons:{
					find:jest.fn(() => { return { matches: true } })
			},
			outputglobalresname:['CP1'],
			outputglobalresvalue:['1']
	}
	let person = {
		personId:123,
		hiddenFields:[{
			fieldname:11
		}]
	};
	let fieldname = 11;
		let wrapper = mount(<IntlProvider locale="en" messages={i18n} >
																		<App {...props}>
																			<Formsy.Form>
																				<ConditionalField dynamicQuestionsForPerson={dynamicQuestionsForPerson}
																				condition={props.condition}
																				personId={props.personId}
																				dynamicQuestions={props.dynamicQuestions}
																				person={props.personId}
																				dynamicQuestionId={props.dynamicQuestionId}
																				hiddenFields={props.hiddenFields[fieldname]}/>
																			</Formsy.Form>
																		</App>
																	</IntlProvider>);
						
							});
});


