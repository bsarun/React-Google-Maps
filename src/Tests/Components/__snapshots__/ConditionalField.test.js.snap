// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptIn Testing conditional field 1`] = `
<div
  className="app-container"
  id="app-container"
>
  <header
    className="main-header has-save-and-exit"
    role="banner"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-3 col-sm-2"
        >
          <h1
            className="logo"
          >
            Covered California
          </h1>
        </div>
        <div
          className="btn-column col-xs-9 col-sm-10 text-right"
        >
          <div
            className="locale-menu en"
          >
            <button
              className="btn btn-toggle hidden-sm hidden-md hidden-lg"
              onClick={[Function]}
            >
              <span
                className="sr-only"
              >
                <span>
                  Open language menu
                </span>
              </span>
            </button>
            <form
              className="locale-options"
              onSubmit={[Function]}
            >
              <div
                className="form-group segmented-control"
              >
                
                
                <div
                  className="controls"
                >
                  <span>
                    <button
                      className="segmented-option btn selected btn-primary"
                      data-value="en"
                      disabled={false}
                      id="locale-options-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                    >
                      <span
                        className="wrapper-text"
                      >
                        English
                      </span>
                    </button>
                  </span>
                  <span>
                    <button
                      className="segmented-option btn btn-default"
                      data-value="es"
                      disabled={false}
                      id="locale-options-1"
                      onBlur={[Function]}
                      onClick={[Function]}
                    >
                      <span
                        className="wrapper-text"
                      >
                        Espaï¿½ol
                      </span>
                    </button>
                  </span>
                </div>
                
                
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    className="main-content"
    role="main"
  >
    <form
      onSubmit={[Function]}
      safeToExit={undefined}
      viewOnlyMode={undefined}
    >
      <span />
    </form>
  </main>
</div>
`;
